### ISSUE TITLE
Unhandled rejection when using "resin.auth.authenticate"

### ISSUE THREAD

**Issue by mkorakakis** [resin-io/resin-sdk#135]
Stacktrace
```
Unhandled rejection ResinRequestError: Request error: Unauthorized
  at C:\Users\korakas\Desktop\test_folder\node_modules\resin-sdk\node_modules\resin-request\build\request.js:129:13
  at tryCatcher (C:\Users\korakas\Desktop\test_folder\node_modules\resin-sdk\node_modules\bluebird\js\main\util.js:26:23)
  at Promise._settlePromiseFromHandler (C:\Users\korakas\Desktop\test_folder\node_modules\resin-sdk\node_modules\bluebird\js\main\promise.js:501:31)
  at Promise._settlePromiseAt (C:\Users\korakas\Desktop\test_folder\node_modules\resin-sdk\node_modules\bluebird\js\main\promise.js:577:18)
  at Promise._settlePromises (C:\Users\korakas\Desktop\test_folder\node_modules\resin-sdk\node_modules\bluebird\js\main\promise.js:693:14)
  at Async._drainQueue (C:\Users\korakas\Desktop\test_folder\node_modules\resin-sdk\node_modules\bluebird\js\main\async.js:123:16)
  at Async._drainQueues (C:\Users\korakas\Desktop\test_folder\node_modules\resin-sdk\node_modules\bluebird\js\main\async.js:133:10)
  at Immediate.Async.drainQueues [as _onImmediate] (C:\Users\korakas\Desktop\test_folder\node_modules\resin-sdk\node_modules\bluebird\js\main\async.js:15:14)
  at processImmediate [as _immediateCallback] (timers.js:367:17)
```

**Comment by jviotti** [comment]
@mkorakakis The issue was caused by `/whoami`. I've send a PR explaining the issue in detail. Can you test to double check it works for you too?

**Comment by jviotti** [comment]
This issue is fixed in v2.6.2.

**Comment by mkorakakis** [comment]
Tested branch jviotti/135/fix/authenticate, didn't encounter the issue.
